### ISSUE TITLE
invalid __shared__ read in moderngpu/src/moderngpu/cta_load_balance.hxx:171

### ISSUE THREAD

**Issue by YuxinxinChen** [gunrock/mini#6]
I compiled pr and run with dataset soc-LiveJournal1.mtx and got the error:
```
cuda-memcheck ./bin/test_pr__x86_64 --file=../../../../gunrock/dataset/large/soc-LiveJournal1/soc-LiveJournal1.mtx

========= Invalid __shared__ read of size 4
=========     at 0x00002010 in /mnt/daisy-mount/mini/gunrock/tests/pr/../../../external/moderngpu/src/moderngpu/cta_load_balance.hxx:171:_ZN4mgpu16launch_box_cta_kINS_12launch_box_tIJNS_7arch_20INS_12launch_cta_tILi128ELi11ELi8ELi0EEENS_7empty_tEEENS_7arch_35INS3_ILi128ELi7ELi5ELi0EEES5_EENS_7arch_52IS4_S5_EEEEEZNS_13lbs_segreduceIS5_ZNS_13lbs_segreduceIS5_ZN7gunrock5oprtr12neighborhood19neighborhood_kernelINSF_2pr12pr_problem_tENSJ_12pr_functor_tEfNS_6plus_tIfEELb0ELb0EEEiSt10shared_ptrIT_ERSO_INSF_10frontier_tIiEEESU_PT1_SV_iRNS_18standard_context_tEEUliiiE_PiPfSN_fJEEEvT0_iSV_iT2_T3_T4_RNS_9context_tEDpT5_EUliiiNS_5tupleIJEEEE_S10_S1B_S11_SN_fJEEEvS12_iSV_iS13_S14_S15_S18_S17_DpT6_EUliiE_JEEEvS12_iDpSV_
=========     by thread (126,0,0) in block (29205,0,0)
=========     Address 0x09015178 is out of bounds
```

It seems a bug in moderngpu

**Comment by yzhwang** [comment]
@YuxinxinChen Hi, Thanks for pointing out. Could you also include the content of the BaseMakefile.mk (only if you have changed it) as well as the GPU you are testing against this? From the error log it could be related to an incorrect setting towards a specific sm version. Of course it could be a moderngpu bug. Let me see more info so that I can better judge.
In summary, a detailed "how to reproduce" would be much appreciated. I'd be happy to fix the bug if it is from moderngpu, as I happen to be reading through his code lately.

**Comment by yzhwang** [comment]
Some update: Tested the current head (the one that fixed the warp sync issue) on a V100 machine with SM_70 setting in the .mk file. All algorithms (including PR and the newly developed k-core) work normally.

**Comment by YuxinxinChen** [comment]
I think we fixed this. Didn't see this problem any more
